server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionServicesImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.objectcomputing.checkins.services.questions;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.List;
+import java.util.UUID;
+
+@Singleton
+public class QuestionServicesImpl implements QuestionServices {
+
+    @Inject
+    private QuestionRepository questionRepository;
+
+    public void setQuestionRepository(QuestionRepository questionRepository) {
+        this.questionRepository = questionRepository;
+    }
+
+    public Question saveQuestion(Question question) {
+
+        List<Question> returnedList = findByValue(question.getText());
+        if (returnedList.size() >= 1) {
+            throw new QuestionDuplicateException("Already exists");
+        }
+        return questionRepository.save(question);
+
+    }
+
+    public List<Question> readAllQuestions() {
+        List<Question> questionList = questionRepository.findAll();
+
+        return questionList;
+
+    }
+
+    public Question findByQuestionId(UUID skillId) {
+
+        Question returned = questionRepository.findByQuestionid(skillId)
+                .orElseThrow(() -> new QuestionNotFoundException("No question for uuid"));
+
+        return returned;
+
+    }
+
+    protected List<Question> findByValue(String text) {
+        List<Question> questionList = null;
+        if(text != null) {
+            questionList = findByText(text);
+        }
```

Comment:
Yes, please...or the equivalent.

Reply:
Well...I guess I can take that back to a degree, as the equivalent appears to be handled in the controller.  It might be worth making this follow a similar implementation pattern still as the others though.